### DISCUSSION TITLE
How Read Path for recent query work - Fanout to all ingesters or few ingesters ?

### DISCUSSION THREAD

**Discussion by rajeshksv** [grafana/mimir#4890] [Help and support] [answered]
For a given user query, does Querier component
a) query all ingesters (or)
b) know which ingester to query and query only those selected nodes ?

If its (b), how does it know what all timeseries are present for matching query/expression

**Comment by charleskorn** [comment, marked as answer]
Hi @rajeshksv, the querier will query all ingesters for the tenant being queried. 

If you have [shuffle sharding](https://grafana.com/docs/mimir/latest/operators-guide/configure/configure-shuffle-sharding/) enabled, this means the querier will query all ingesters in the tenant's shuffle shard, as only those ingesters could have series for the tenant. 

If shuffle sharding is disabled, then all ingesters are queried as all ingesters could have series for the tenant.
